api/include/opentelemetry/context/context.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <map>
+#include <string>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace context
+{
+
+
+/*The context class provides a context identifier*/
+class Context{
+  private:
+    /*The identifier itself*/
+    std::map<std::string, int> ctx_;
+
+  public:
```

Comment:
Things to be considered:
1. Is this an abstract class or the actual implementation? If it is the former, need to have pure virtual functions.
2. Will we have APIs to Create a Key, Get Value and Set Value https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/context/context.md#create-a-key.